kratos/solving_strategies/builder_and_solvers/residualbased_block_builder_and_solver.h

```diff
@@ -766,8 +766,11 @@ class ResidualBasedBlockBuilderAndSolver
         }
         if (Dx.size() != BaseType::mEquationSystemSize)
             Dx.resize(BaseType::mEquationSystemSize, false);
-        if (b.size() != BaseType::mEquationSystemSize)
+        TSparseSpace::SetToZero(Dx);

```

Comment:
this should be 

if(...){
   resize
   SetToZero
}

other than that i agree (same for the b)

Reply:
IMO may be initialized but filled of values different of zero, therefore may be set to zero always

Reply:
ok

Reply:
I agree with Vicente, even if it has the right size, it might still be uninit